File: libs/sdk-bindings/bindings-android/lib/src/main/kotlin/breez_sdk_notification/ServiceConfig.kt
package breez_sdk_notification

data class ServiceConfig(
    /** Channel setup fee limit (in msats) */
    val channelFeeLimitMsat: ULong = defaultChannelFeeLimitMsat,
) {
    companion object {
        /** Channel setup fee limit (in msats) for the default ServiceConfig */
        const val defaultChannelFeeLimitMsat: ULong = 0u

        /** The default ServiceConfig */
        fun default(): ServiceConfig = ServiceConfig(defaultChannelFeeLimitMsat)
    }
}

File: libs/sdk-bindings/bindings-swift/Sources/BreezSDK/ServiceConfig.swift
public struct ServiceConfig {
    // MARK: - Constants
    public struct Constants {
        /// Channel setup fee limit (in msats)  for the default instance of ServiceConfig
        public static let defaultChannelFeeLimitMsat = UInt64(0)
    }
    
    // MARK: - Default instance
    /// The default ServiceConfig object
    public static let `default`: ServiceConfig = ServiceConfig(channelFeeLimitMsat: ServiceConfig.Constants.defaultChannelFeeLimitMsat)
    
    // MARK: - Properties
    /// Channel setup fee limit (in msats)
    public var channelFeeLimitMsat: UInt64! = UInt64(0)
    
    // MARK: - Life Cycle
    public init(channelFeeLimitMsat: UInt64) {
        self.channelFeeLimitMsat = channelFeeLimitMsat
    }
}
